mix.exs

```diff
@@ -38,9 +38,9 @@ defmodule Phoenix.GenSocketClient.Mixfile do
       {:poison, "~> 2.0 or ~> 3.0", optional: true},
       {:phoenix, "~> 1.3", only: :test},
       {:cowboy, "~> 1.0", only: :test},
-      {:credo, "~> 0.8.10", runtime: false},
-      {:dialyze, "~> 0.2.1", only: :dev},
-      {:ex_doc, "~> 0.17.1", only: :docs}
+      {:credo, "~> 0.8.10", only: [:dev, :test], runtime: false, optional: true},
+      {:dialyze, "~> 0.2.1", only: :dev, optional: true},
+      {:ex_doc, "~> 0.17.1", only: :docs, optional: true}
```

Comment:
I don't think we need `optional: true` in these three dependencies. As soon as dep is not included in `prod` env, it's not going to be fetched as a transitive dep. So for example, if you currently depend on the latest published `phoenix_gen_socket_client`, neither`dialyze` nor `ex_doc` are going to be pulled when invoking `mix deps.get` in your project, despite the fact that they aren't listed with `optional: true` setting in the latest release.